### ISSUE TITLE
Early stopping

### ISSUE THREAD

**Issue by kadzaki** [prdwb/bert_hae#7]
Hi !
First, thank you for this great model! i'am wondering how can i add early stopping support in your model? I have a training set, a validation set and a test set, I want to train on the training set, do early stopping with the validation set and get predictions on the test set.

**Comment by prdwb** [comment]
Since this model does training and evaluation in an alternating way, I guess you can check the validation F1 at the end of each evaluation stage and decide whether to stop training. See https://github.com/prdwb/bert_hae/blob/d5fc14d68a754f7056658cbdc4fc34938acf5a2c/hae.py#L336
Thanks.

**Comment by kadzaki** [comment]
OK thanks!
